Makefile.am

```diff
@@ -158,14 +158,66 @@ base_files = \
 bin_PROGRAMS += csd
 
 csd_SOURCES = \
-	${base_files} \
+	${BASE_FILES} \
```

Comment:
do these files change behavior in compile time according to the binary it's compiling to?  It looks like you want a convenience library so you don't build all these ${BASE_FILES}  over and over again